quarkus/pom.xml

```diff
@@ -74,8 +74,14 @@
                 <version>${quarkus.version}</version>
                 <type>pom</type>
                 <scope>import</scope>
+                <exclusions>
+                    <exclusion>  <!-- declare the exclusion here -->
+                        <groupId>org.apache.maven</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
```

Comment:
"vulnerability scanners running on preview versions of keycloak-x are complaining about the inclusion of maven into the tar.gz artifact."

Which vulnerability scanners are you referring to? This doesn't actually change anything about what dependencies are included at all, as it's just the dependency-management section. So, to be this seems to be an issue in the vulnerability scanner as it's trying to be a bit to overly smart here.

Reply:
snyk... the ban of my existence atm... ;-) 

since I raised the PR I suspect the issue is being caused by the dist-image still including `/opt/jboss/keycloak/lib/lib/deployment/`. Since opening the PR I realises that quarkus allows us to remove the deployments directory as long the use is happy with forfeiting re-augmentation. What that means I still need to discover. :-) 

(pls note /opt/jboss/keycloak is essentially where we copy `keycloak.x-15.0.2` into)

```
$ ls -l /opt/jboss/keycloak/lib/lib/**/*maven*
-rw-r--r-- 1 jboss jboss 106171 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/io.quarkus.quarkus-bootstrap-maven-resolver-1.13.3.Final.jar
-rw-r--r-- 1 jboss jboss  57824 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-artifact-3.6.3.jar
-rw-r--r-- 1 jboss jboss  14262 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-builder-support-3.6.3.jar
-rw-r--r-- 1 jboss jboss 633028 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-core-3.6.3.jar
-rw-r--r-- 1 jboss jboss  98096 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-embedder-3.6.3.jar
-rw-r--r-- 1 jboss jboss 215412 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-model-3.6.3.jar
-rw-r--r-- 1 jboss jboss 192302 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-model-builder-3.6.3.jar
-rw-r--r-- 1 jboss jboss  47035 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-plugin-api-3.6.3.jar
-rw-r--r-- 1 jboss jboss  27158 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-repository-metadata-3.6.3.jar
-rw-r--r-- 1 jboss jboss  66242 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-resolver-provider-3.6.3.jar
-rw-r--r-- 1 jboss jboss  44047 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-settings-3.6.3.jar
-rw-r--r-- 1 jboss jboss  42090 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.maven-settings-builder-3.6.3.jar
-rw-r--r-- 1 jboss jboss 148983 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.resolver.maven-resolver-api-1.4.1.jar
-rw-r--r-- 1 jboss jboss  44143 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.resolver.maven-resolver-connector-basic-1.4.1.jar
-rw-r--r-- 1 jboss jboss 180696 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.resolver.maven-resolver-impl-1.4.1.jar
-rw-r--r-- 1 jboss jboss  36732 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.resolver.maven-resolver-spi-1.4.1.jar
-rw-r--r-- 1 jboss jboss  31013 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.resolver.maven-resolver-transport-wagon-1.4.1.jar
-rw-r--r-- 1 jboss jboss 167529 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.resolver.maven-resolver-util-1.4.1.jar
-rw-r--r-- 1 jboss jboss 166562 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.shared.maven-shared-utils-3.2.1.jar
-rw-r--r-- 1 jboss jboss  11586 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.wagon.wagon-file-3.3.4.jar
-rw-r--r-- 1 jboss jboss  11367 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.wagon.wagon-http-3.3.4.jar
-rw-r--r-- 1 jboss jboss  41123 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.wagon.wagon-http-shared-3.3.4.jar
-rw-r--r-- 1 jboss jboss  55776 Aug 20 07:27 /opt/jboss/keycloak/lib/lib/deployment/org.apache.maven.wagon.wagon-provider-api-3.3.4.jar
```

BTW, I appreciate sometimes the situation is outside Keycloak's control.

Challenge is: With dependency related security incident becoming far more common, the "vulnerability management" ecosystem is out of control. Even Sonatype is now issuing "security advisories. So it is not suprise that in the race for "detecting more vulnerabilities, vendor are going as far as listing H2, a DB not designed for production use, as a vulnerability.

The amount of time spent managing senior management expectations is beyond surreal.

Reply:
as I expected :-( 

```
bash-4.2$ rm -rf /opt/jboss/keycloak/lib/lib/deployment
bash-4.2$ /opt/jboss/keycloak/bin/kc.sh config  --db=mysql
...
Updating the configuration and installing your custom providers, if any. Please wait.
ERROR: Failed to update server configuration.
ERROR: /opt/jboss/keycloak/lib/lib/deployment/deployment-class-path.dat
```

Since my images are used on deployments using different DBs, I may need to move from single docker to 3 images: 

`base` (i.e. no `config`) 
`db1` (e.g `config --db=mysql`)
`db2` (e.g. `config --db=mariadb`)

Running config with k8s is not an option as they vuln scanners test both on running instances and images. :-\

Reply:
Honestly not following what you're saying. However, if snyk is complaining that maven is being included that's invalid. There's libraries from Maven sure, and the word "maven" and "jar" is in the deployment. However, that doesn't make it a valid issue.

The change you are making here is only removing the dependencies from the maven dependency management section, which doesn't mean they won't be included, just means you don't inherit the version from the Quarkus BOM.

Whether or not we can actually remove these dependencies I'm not totally sure. @pedroigor @DGuhr ?

Reply:
I think we can. In fact, after the last few changes quite a few dependencies were included in the distribution without any need for them.

We discussed some time ago to review those deps, but as we did more changes we did not track this particular aspect of Dist.X.

Shall we create a more generic issue too to review libs before we release?

Reply:
@trixpan Can you please trying this approach. Change the `application.properties` within the runtime module and include this config:

```
quarkus.class-loading.removed-artifacts=org.junit.jupiter:junit-jupiter,org.junit.jupiter:junit-jupiter-api,org.apache.maven:maven-artifact
```

There you should add any dependency that does not make sense such as those plus maven, testcontainers, etc. I don't think removing maven should harm because re-augmentation should be based on the application metadata created when we build the disttribution.